pydantic/config.py

```diff
@@ -62,8 +62,11 @@ class BaseConfig:
     json_encoders: Dict[Union[Type[Any], str, ForwardRef], AnyCallable] = {}
     underscore_attrs_are_private: bool = False
 
-    # whether inherited models as fields should be reconstructed as base model
+    # whether inherited models as fields should be reconstructed as base model,
+    # and whether such a copy should be shallow or deep
     copy_on_model_validation: bool = True
+    copy_on_model_validation_shallow: bool = False
```

Comment:
If we want this, I would rather switch to an enum for
`copy_on_model_validation` with options `'disable', 'shallow', 'deep'`

Reply:
Or just a literal?

Reply:
string literal better than enum I would think